runtime/lua/vim/treesitter.lua

```diff
@@ -506,6 +506,26 @@ function M.inspect_tree(opts)
   require('vim.treesitter.playground').inspect_tree(opts)
 end
 
+--- Determines diagnostics for a buffer containing treesitter queries (|treesitter-query|).
+--- For runtime query files that target installed treesitter parsers, this function will use
+--- treesitter API to determine admissible identifiers to describe nodes of those parsers and
+--- verifies that top-level s-expressions can be loaded in isolation by Neovim to debug errors
+--- in runtime query files. The found diagnostics are reported using |diagnostic-api|.
+--- @param buf integer Buffer to lint
+--- @param opts nil|table Options for linting
+--- - langs nil|string|string[] One or more parsers which the current queries target (see |treesitter-parsers|)
+---   or a function that determines the list of parsers given `buf`, `opts`. By default the query parser is determined
+---   using the containing folder of the query buffer (`<lang>/<query_type>.scm`, e.g. `cpp` for `cpp/highlights.scm`)
+function M.query_linter_lint(buf, opts)
+  require('vim.treesitter._query_linter').lint(buf, opts)
+end
+
+--- Clears diagnostics of the query linter for a given buffer
+--- @param buf integer Buffer to clear diagnostics
+function M.query_linter_clear(buf)
+  require('vim.treesitter._query_linter').clear(buf)
+end
```

Comment:
I suggest moving these to `vim/treesitter/query.lua` and renaming to `lint`. Does `clear` need to be a separate function? Why not a "keyword argument" in the `opts` table?

```
vim.treesitter.query.lint(buf, {})
vim.treesitter.query.lint(buf, {clear = true})
```

Reply:
💯

Reply:
maybe `action = "clear"` with default of action being "lint".

Reply:
No, I think @gpanders suggestion is fine (we have a similar pattern for creating autocommand groups). If in doubt, keep it simple :)